src/main/java/seedu/address/model/loan/LoanTime.java

```diff
@@ -35,133 +32,203 @@
+        } else {
+            // The string only contains the Time. Check if the Time is valid.
+            return isValidTime(objString);
+        }
     }
 
     /**
-     * Returns if a given string is a valid long LoanTime.
+     * Checks if a given string is a valid long LoanTime.
+     *
+     * @param test The string to test
+     * @return true if the string is a valid long LoanTime
      */
     public static boolean isValidLongLoanTimeFormat(String test) {
         return test.matches(LONG_LOANTIME_VALIDATION_REGEX);
     }
 
     /**
-     * Returns if a given string is a valid short LoanTime.
+     * Checks if a given string is a valid short LoanTime.
+     *
+     * @param test The string to test
+     * @return true if the string is a valid short LoanTime
      */
     public static boolean isValidShortLoanTimeFormat(String test) {
         return test.matches(SHORT_LOANTIME_VALIDATION_REGEX);
     }
 
     /**
-     * Returns if a given string is a valid Date.
-     * If there is a better way of doing this without relying on thrown exceptions
+     * Checks if a given string is a valid Date.
+     *
+     * @param test The string to test
+     * @return true if the string is a valid Date
      */
     public static boolean isValidDate(String test) {
-        DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
-
-        sdf.setLenient(false);
-
         try {
-            sdf.parse(test);
-        } catch (ParseException e) {
+            DEFAULT_DATE_FORMATTER.parse(test);
+        } catch (DateTimeParseException e) {
             return false;
         }
+
         return true;
     }
 
     /**
-     * Returns if a given string is a valid Time.
+     * Checks if a given string is a valid Time.
+     *
+     * @param test The string to test
+     * @return true if the string is a valid Time
      */
     public static boolean isValidTime(String test) {
         String[] timeData = test.split(":");
-        if (timeData.length <= 1) {
-            return false;
-        }
+        assert timeData.length == 2;
+
         return (Integer.parseInt(timeData[0]) < 24) && (Integer.parseInt(timeData[1]) < 60);
     }
 
     /**
      * Returns the difference in time given one other LoanTime object.
-     * This will be returned as a long of number of second.
+     * This will be returned as a long of number of minutes.
      * Function returns 0 if the specified other LoanTime is before the current LoanTime.
      *
      * @param otherTime LoanTime object to be compared to.
      */
     public long loanTimeDifferenceMinutes(LoanTime otherTime) {
-        long timeDifference = (otherTime.value.toEpochMilli() - this.value.toEpochMilli());
-        return (timeDifference >= 0) ? timeDifference / 60000 : 0;
+        long timeDifference = Duration.between(this.value, otherTime.value).toMinutes();
+        return (timeDifference >= 0) ? timeDifference : 0;
     }
```

Comment:
This could re-use the static method following it, to reduce repeated code.

Reply:
That's true. I'll make that change.